megablocks/layers/moe.py

```diff
@@ -200,6 +204,11 @@ def permute_and_compute(
             x, indices, expert_weights, bins, top_k)
 
     def forward_once(self, x, expert_weights, top_experts):
+        # Expert dropout: Randomly drop experts based on the dropout rate
```

Comment:
I think that you could add this in the `forward` function and avoid duplicating it for both `forward_once` and `parallel_forward_once`?